pkg/otp/otp.go

```diff
@@ -45,47 +45,43 @@ func Calculate(name string, sec gopass.Secret) (twofactor.OTP, string, error) {
 	}
 
 	if strings.HasPrefix(secKey, "otpauth://") {
-		return twofactor.FromURL(secKey) //nolint:wrapcheck
+		debug.Log("parsing otpauth:// URL %q", out.Secret(secKey))
+
+		k, err := otp.NewKeyFromURL(secKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse otpauth URL: %w", err)
+		}
+
+		return k, nil
 	}
 
-	otp, err := twofactor.NewGoogleTOTP(twofactor.Pad(secKey))
+	debug.Log("assembling otpauth URL from secret only (%q), using defaults", out.Secret(secKey))
+
+	// otpauth://totp/Example:alice@google.com?secret=JBSWY3DPEHPK3PXP&issuer=Example
+	key, err := otp.NewKeyFromURL(fmt.Sprintf("otpauth://totp/new?secret=%s&issuer=gopass", secKey))
```

Comment:
This is not supporting HOTP, we might want to errror out if HOTP was requested, no?

Reply:
How would we know that HOTP was requested?

Reply:
The URL changes, it says "otpauth://hotp" instead.
Or we use the KV `hotp` instead of `otpauth` or `totp`.

Also I guess just relying on the library's `Type` method should be enough.

Reply:
Actually this is only the fallback. If we end up having only the secret we can only assume that it's TOTP. Everything else would require information we just don't have.

Maybe it's not worth keeping this fallback after all. But I guess somebody would complain if we drop it.